build/common/installer/scripts/tomlparser-geneva-config.rb

```diff
@@ -0,0 +1,130 @@
+#!/usr/local/bin/ruby
+
+require "tomlrb"
+
+require_relative "ConfigParseErrorLogger"
+
+@configMapMountPath = "./etc/config/settings/agent-settings"
+@configSchemaVersion = ""
+
+# configmap settings related to geneva logs config
+@genevaEnvironment = ""
+@genevaAccount = ""
+@genevaNamespace = ""
+@genevaConfigVersion = ""
+@genevaAuthId = ""
+GENEVA_SUPPORTED_ENVIRONMENTS = ["Test", "Stage", "DiagnosticsProd", "FirstpartyProd", "BillingProd", "ExternalProd", "CaMooncake", "CaFairfax", "CaBlackforest"]
+
+# Use parser to parse the configmap toml file to a ruby structure
+def parseConfigMap
+  begin
+    # Check to see if config map is created
+    if (File.file?(@configMapMountPath))
+      puts "config::configmap container-azm-ms-agentconfig for agent settings mounted, parsing values"
+      parsedConfig = Tomlrb.load_file(@configMapMountPath, symbolize_keys: true)
+      puts "config::Successfully parsed mounted config map"
+      return parsedConfig
+    else
+      puts "config::configmap container-azm-ms-agentconfig for agent settings not mounted, using defaults"
+      return nil
+    end
+  rescue => errorStr
+    ConfigParseErrorLogger.logError("Exception while parsing config map for agent settings : #{errorStr}, using defaults, please check config map for errors")
+    return nil
+  end
+end
+
+# Use the ruby structure created after config parsing to set the right values to be used as environment variables
+def populateSettingValuesFromConfigMap(parsedConfig)
+  begin
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil?
+      
+      geneva_logs_config = parsedConfig[:agent_settings][:geneva_logs_config]
+      if !geneva_logs_config.nil?
+        puts "config: parsing geneva_logs_config settings"
+        genevaEnvironment = geneva_logs_config[:environment]
+        genevaAccount = geneva_logs_config[:account]
+        genevaNamespace = geneva_logs_config[:namespace]
+        genevaConfigVersion = geneva_logs_config[:configversion]
+        if !genevaEnvironment.nil? && !genevaAccount.nil? && !genevaNamespace.nil? && !genevaConfigVersion.nil?
+          if GENEVA_SUPPORTED_ENVIRONMENTS.include?(genevaEnvironment)
```

Comment:
this map needs to include environment for airgap clouds or this check can be removed  and we can ask the 1P customers to follow geneva docs for the environment names.